tests/test_gates.py

```diff
@@ -208,6 +209,43 @@ def output(input):
     assert np.allclose(unitary, unitary_expected)
 
 
+@pytest.mark.parametrize("op_name", ld_operations)
+def test_compare_sparse_and_dense_operations(op, op_name):
+    """Test if lightning device correctly applies QubitUnitary sparse operators by comparing it with the dense case."""
+    if device_name != "lightning.qubit":
+        pytest.skip("Skipping tests if not lightning.qubit.")
+
+    if op_name in ("BasisState", "StatePrep"):
+        pytest.skip("Skipping operation because it is a state preparation")
+    if op == None:
+        pytest.skip("Skipping operation.")
+
+    # op is a tuple with the operation, its parameters, and its keyword arguments
+    qml_op = op[0](*op[1], **op[2])
+    wires = op[2]["wires"]
+    num_wires = len(wires)
+    matrix = qml.matrix(qml_op)
+
+    dev = qml.device(device_name, wires=num_wires)
+
+    @qml.qnode(dev)
+    def circuit_dense(input):
+        qml.BasisState(input, wires=range(num_wires))
+        qml.QubitUnitary(matrix, wires=wires)
+        return qml.state()
+
+    @qml.qnode(dev)
+    def circuit_sparse(input):
+        qml.BasisState(input, wires=range(num_wires))
+        qml.QubitUnitary(csr_matrix(matrix), wires=wires)
+        return qml.state()
```

Comment:
If you have time, to keep same style as controlled
```suggestion

    sparse_matrix = csr_matrix(matrix)

    @qml.qnode(dev)
    def circuit_sparse(input):
        qml.BasisState(input, wires=range(num_wires))
        qml.QubitUnitary(sparse_matrix, wires=wires)
        return qml.state()
```

Reply:
Let me know if it is good now